### PR TITLE
Archive Old Forms Script

### PR DESCRIPTION
**Pull request by zandre-eng**
## Technical Summary

Link to ticket [here](https://dimagi-dev.atlassian.net/browse/SC-2751).

This PR is not intended to be merged. It simply exists as a way in which feedback can be given from other devs regarding a script that will be run on production.

The purpose of this script is to archive forms for cases that have been closed or predate 1 May 2022. The target domain is `fmoh-echis-staging`. Due to the size of relevant cases (about 2.3 million), the script will apply batching.

### CODE REVIEW

File: corehq/util/archive_old_forms.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+batch_count = ceil(total_cases / BATCH_SIZE)
+print(f"Total of {total_cases} cases to process over {batch_count} batches")
+
+def process_batch(batch_number):
+    start_count = batch_number * BATCH_SIZE
+    percent_complete = int(start_count / total_cases)
+    print(f"Processing {start_count}-{BATCH_SIZE * (batch_number + 1)} ({percent_complete}%)")
+
+    # Get all relevant cases
+    closed_or_old_case_ids = (
```
Comment (kaapstorm) [inline corehq/util/archive_old_forms.py:38]: I would add `.sort("opened_on")` to ensure that each chunk is contiguous.

File: corehq/util/archive_old_forms.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# Note: There are over 2.3 million cases that need to be archived on the target project. For this reason,
+# batching will be necessary. Furthermore, the batch sizes are kept small as each case can have 2-3+ forms
+# that will need to be archived.
+
+# Constants
+DOMAIN = 'test-65' # 'fmoh-echis-staging'
+DATE_CONSTRAINT = datetime.date(2022, 5, 1)
+BATCH_SIZE = 100
+ERROR_LOG_FILE_PATH = os.path.expanduser("~/zeng_script_error.log")
+SUCCESS_LOG_FILE_PATH = os.path.expanduser("~/zeng/script_success.log")
```
Comment (kaapstorm) [inline corehq/util/archive_old_forms.py:17]: Are you intentionally putting this file in a subdirectory, and the other log file in the home directory?

**Review by kaapstorm — COMMENTED** [review]
This looks good to me.

File: corehq/util/archive_old_forms.py (hunk 1 above)
Comment (zandre-eng) [reply corehq/util/archive_old_forms.py:38]: That is a good idea :+1:

File: corehq/util/archive_old_forms.py (hunk 2 above)
Comment (zandre-eng) [reply corehq/util/archive_old_forms.py:17]: That seems to be a missing slash, they should both be going to the `zeng` subdirectory.

**Pull request closed without merging**